holoviews/plotting/mpl/chart.py

```diff
@@ -288,9 +295,15 @@ def _process_hist(self, hist):
         edges = hist.interface.coords(hist, x, edges=True)
         values = hist.dimension_values(1)
         hist_vals = np.array(values)
+        xlim = hist.range(0)
+        ylim = hist.range(1)
+        isdatetime = False
+        if edges.dtype.kind == 'M' or isinstance(edges[0], datetime_types):
+            edges = date2num([v.tolist() if isinstance(v, np.datetime64) else v for v in edges])
+            xlim = date2num([v.tolist() if isinstance(v, np.datetime64) else v for v in xlim])
+            isdatetime = True
         widths = np.diff(edges)
-        lims = hist.range(0) + hist.range(1)
-        return edges[:-1], hist_vals, widths, lims
+        return edges[:-1], hist_vals, widths, xlim+ylim, isdatetime
```

Comment:
I can still conceive of wanting support for other numeric types one day (e.g rationals?) but this is fine for now. We can use a general type specifier when we find ourselves needing to do something similar for another type. This this fine for now.